docs/modules/qbi-helper/interest-template.md

```diff
@@ -0,0 +1,146 @@
+<%(TOC)%>
+
+# How to use InterestTemplate
+
+<%(Callout message="
+Before reading this document, make sure you have read:
+
+  * [Introduction to Query-based interest]({{urlRoot}}/modules/qbi-helper/intro-to-qbi)
+")%>
+
+An [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) is a wrapper around the `Interest` component, providing intuitive methods to add, replace and clear queries from the underlying interest map.
+
+## Create a template
+
+You create an empty [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) by calling `InterestTemplate.Create()`. This is useful when defining a set of interest queries for the first time, for example when defining Entity Templates.
+
+```csharp
+var basicInterestTemplate = InterestTemplate.Create();
+```
+
+You can also construct this from an existing [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template) or set of interest queries. This creates a deep copy of the [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template), to allow you to modify the queries without affecting the original set.
+
+```csharp
+var advancedInterestTemplate = InterestTemplate.Create(basicInterestTemplate);
+```
+
+## Modify a template
+
+When modifying an [`InterestTemplate`]({{urlRoot}}/api/query-based-interest/interest-template), you must specify which particular component's queries you are modifying. You can either:
+
+* provide the component as a type argument
+* pass the component ID as the first argument
+
+For example, to add a query you might do one of the following:
+
+```csharp
+// Component identified by type argument
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Component identified by ID given as first argument
+InterestTemplate.Create()
+    .AddQueries(Position.ComponentId, query1, query2);
+```
+
+### Add
+
+When adding queries, you can either provide the queries as parameters or an enumerable set:
+
+```csharp
+// Parameters
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(query1, query2);
+
+// Enumerable
+var queryList = new List() { query1, query2 };
+InterestTemplate.Create()
+    .AddQueries<Position.Component>(queryList);
+```
+
+### Replace
+
+This operation replaces all the existing queries for the given component ID with the new queries you pass in. In a similar way to adding queries, you can provide the queries as parameters or an enumerable set.
+
+```csharp
+// Parameters
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(query1);
+
+// Enumerable
+var queryList = new List() { query1 };
+InterestTemplate.Create(basicInterestTemplate)
+    .ReplaceQueries<Position.Component>(queryList);
+```
+
+> At least one query must be provided, otherwise the existing queries for a given authoritative component will not be removed.
```

Comment:
```suggestion
> At least one query must be provided, otherwise this operations does not remove the existing queries for a given authoritative component.
```